OverlayPlugin.Core/MemoryProcessors/Combatant/LineCombatant.cs

```diff
@@ -0,0 +1,411 @@
+                if (lastUpdatedDiff > delayTransformationID && combatant.TransformationId != oldCombatant.TransformationId)
+                {
+                    changed.Add(combatant.GetType().GetField("TransformationId"));
+                }
+                var delayWeaponID = inCombat ? CombatantChangeCriteria.InCombatDelayWeaponID : CombatantChangeCriteria.OutOfCombatDelayWeaponID;
+                if (lastUpdatedDiff > delayWeaponID && combatant.WeaponId != oldCombatant.WeaponId)
+                {
+                    changed.Add(combatant.GetType().GetField("WeaponId"));
+                }
+                var delayTargetID = inCombat ? CombatantChangeCriteria.InCombatDelayTargetID : CombatantChangeCriteria.OutOfCombatDelayTargetID;
+                if (lastUpdatedDiff > delayTargetID && combatant.TargetID != oldCombatant.TargetID)
+                {
+                    changed.Add(combatant.GetType().GetField("TargetID"));
+                }
+                var delayPosition = inCombat ? CombatantChangeCriteria.InCombatDelayPosition : CombatantChangeCriteria.OutOfCombatDelayPosition;
+                if (lastUpdatedDiff > delayPosition)
+                {
+                    // This check seems redundant but it's less expensive than the check below against distance
+                    // so it uses less CPU
+                    if (combatant.PosX != oldCombatant.PosX || combatant.PosY != oldCombatant.PosY || combatant.PosZ != oldCombatant.PosZ)
+                    {
+                        var dist = Math.Sqrt(
+                            Math.Pow(combatant.PosX - oldCombatant.PosX, 2)
+                            + Math.Pow(combatant.PosY - oldCombatant.PosY, 2)
+                            + Math.Pow(combatant.PosZ - oldCombatant.PosZ, 2));
+                        var checkDist = inCombat ? CombatantChangeCriteria.InCombatDistancePosition : CombatantChangeCriteria.OutOfCombatDistancePosition;
+                        if (dist > checkDist)
+                        {
+                            if (combatant.PosX != oldCombatant.PosX)
+                            {
+                                changed.Add(combatant.GetType().GetField("PosX"));
+                            }
+                            if (combatant.PosY != oldCombatant.PosY)
+                            {
+                                changed.Add(combatant.GetType().GetField("PosY"));
+                            }
+                            if (combatant.PosZ != oldCombatant.PosZ)
+                            {
+                                changed.Add(combatant.GetType().GetField("PosZ"));
+                            }
+                        }
+                    }
+                    else if (combatant.Heading != oldCombatant.Heading)
+                    {
+                        var diff = Math.Abs((combatant.Heading + Math.PI) - (oldCombatant.Heading + Math.PI));
```

Comment:
Adding `Math.PI` prevents weird issues with negative numbers, I think. Though it might not be needed now, this comparison was a bit more odd previously.

Reply:
I guess my thought is that even if one or both of `h1` and `h2` are negative, `Math.abs(h1 - h2)` is the absolute difference between them.